### PR TITLE
[torchlib] Fix aten::arange to support dynamic shapes

### PR DESCRIPTION
**Pull request by titaiwangms**
Runinng SmolLM_1_7b with dynamic shapes discovered that arange.start was not dynamic anymore, since https://github.com/microsoft/onnxscript/pull/1781/files.

### CODE REVIEW

**Review by justinchuby — APPROVED** [review]
Thanks, also interesting - does this mean we need to update https://github.com/pytorch/pytorch/blob/10f16cc7daf89a73c1177dfbd0bb0fbb0fc8683b/aten/src/ATen/native/native_functions.yaml#L780 ? Should we create an issue for propose a fix?

**Comment by justinchuby** [comment]
Should we also fix the rest of the arange overloads?

**Comment by titaiwangms** [comment]
> Should we also fix the rest of the arange overloads?

Sure, I can update the whole family. And record the question that does Scalar cover sym_int and sym_float in Meta doc.

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 1]
```diff
@@ -599,66 +595,65 @@ def aten_arange_start(
...
+    end: TRealUnlessFloat16OrInt8,
+    step: TRealUnlessFloat16OrInt8,
+) -> Tuple[FLOAT, FLOAT, FLOAT]:
+    zero = op.Cast(0.0, to=FLOAT.dtype)
+    start = op.Cast(start, to=FLOAT.dtype)
+    end = op.Cast(end, to=FLOAT.dtype)
+    step = op.Cast(step, to=FLOAT.dtype)
+
+    start = op.Where(op.Less(start, zero), op.Ceil(start), start)
+    start = op.Where(op.Less(step, zero), op.Floor(start), start)
```
Comment (titaiwangms) [inline onnxscript/function_libs/torch_lib/ops/core.py:617]: @justinchuby This is unavoidably brought back. Were you trying to solve a performance issue with https://github.com/microsoft/onnxscript/pull/1781/files perhaps?

Comment (xadupre) [reply onnxscript/function_libs/torch_lib/ops/core.py:617]: I wonder why it is needed to fix start paremeter?

**Comment by xadupre** [comment]
Is it possible to add a unit test to make sure there is no regression about this?

**Comment by titaiwangms** [comment]
> Is it possible to add a unit test to make sure there is no regression about this?

I will merge this first to unblock the model, and then come up with tests as follow-up.

File: onnxscript/function_libs/torch_lib/ops/core.py (hunk 1 above)
Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/ops/core.py:617]: I remember there were test cases that would break without this logic

Comment (titaiwangms) [reply onnxscript/function_libs/torch_lib/ops/core.py:617]: I am reverting the whole changes from [https://github.com/microsoft/onnxscript/pull/1781](https://github.com/microsoft/onnxscript/pull/1781/files)

**Comment by justinchuby** [comment]
The old arange logic had issues with inconsistent types. Please make sure that's resolved. I am happy if you fix one and take time to ensure the rest is correct in a separate PR, if that allows faster progress

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 2]
```diff
@@ -599,66 +595,63 @@ def aten_arange_start(
...
 
 
 @torch_op("aten::arange.start_step", trace_only=True)
 def aten_arange_start_step(
-    start: float,
-    end: float,
-    step: float = 1.0,
+    start: TRealUnlessFloat16OrInt8,
+    end: TRealUnlessFloat16OrInt8,
+    step: TRealUnlessFloat16OrInt8 = 1.0,
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/core.py:626]: does this default work?

Comment (titaiwangms) [reply onnxscript/function_libs/torch_lib/ops/core.py:626]: Add issue https://github.com/microsoft/onnxscript/issues/1914 to follow up on TracedONNXFunction extension.

**Pull request merged**